libyul/backends/evm/OptimizedEVMCodeTransform.cpp

```diff
@@ -497,7 +497,133 @@ void OptimizedEVMCodeTransform::operator()(CFG::BasicBlock const& _block)
 			CFG::BuiltinCall const* builtinCall = get_if<CFG::BuiltinCall>(&_block.operations.back().operation);
 			yulAssert(builtinCall, "");
 			yulAssert(builtinCall->builtin.get().controlFlowSideEffects.terminatesOrReverts(), "");
-		}
+		},
+		[&](CFG::BasicBlock::Switch const& _switch)
+		{
+			std::cout << "In OptimizedEVMCodeTransform: creating switch statement" << std::endl;
```

Comment:
You will need
```
		   createStackLayout(debugDataOf(_switch), blockInfo.exitLayout);
```
here, which will be the optimal layout from which all the switch cases can be generated cheaply below.

Reply:
Added in [e8a3f92](https://github.com/ethereum/solidity/pull/12978/commits/e8a3f92578904dddbabacdb86f4c88ebe6a8440f)